### PR TITLE
Client redesign

### PR DESCRIPTION
**Pull request by alir128**
image
image
image

#92

All 6 issues mentioned before are now resolved.

### CODE REVIEW

**Comment by Lakshya2610** [comment]
@alir128 I don't know if you realized it, but pausing the audio and replaying adds latency between src and client. While that itself is not a problem, WebRTC tries to catch up to src by speeding up the audio. So, do we want this?

File: client-redesign/src/router/index.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  {
+    path: "/about",
+    name: "about",
+    // route level code-splitting
+    // this generates a separate chunk (about.[hash].js) for this route
+    // which is lazy-loaded when the route is visited.
+    component: () =>
+      import(/* webpackChunkName: "about" */ "../views/About.vue")
+  },
+  { path: "/:id", component: User }
```
[the remark this answers is not in the document]
Comment (alir128) [reply client-redesign/src/router/index.js]: I can rename :id to :room, but toonin through url by entering the room name is working as shown in the screenshot also that I posted.

File: client-redesign/src/components/roomName.vue [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    }
+  },
+  components: {
+    Button
+  },
+  methods: {
+    toonin() {
+      this.connectToRoom();
+    },
+    connectToRoom() {
```
[the remark this answers is not in the document]
Comment (alir128) [reply client-redesign/src/components/roomName.vue]: Do you want me to move the RTC logic to main.vue, I can’t move it the store as it doesn’t have access to socket io and so it became complicated to subscribe to socket events. And no state management isn’t actually taking place here only Vuex Actions are being called, which call mutations inside the store which is how it mentions to do in Vuex documentation. Component methods -> Vuex actions -> Vuex Mutations -> State updated -> getters used to map state back inside component.

File: client-redesign/src/views/User.vue [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import connectionStatus from "@/components/connection.vue"
+import PlayerControls from "@/components/media.vue"
+import Title from "@/components/title.vue"
+export default {
+  name: "user",
+  data: () => ({
+    volume: 50
+  }),
+  components: {
+    card,
```
[the remark this answers is not in the document]
Comment (alir128) [reply client-redesign/src/views/User.vue]: Sure will address this.

**Comment by alir128** [comment]
> @alir128 UI looks fantastic! There's only one problem though. I don't know if you realized it, but pausing the audio and replaying adds latency between source and client. While that itself is not a problem, WebRTC tries to catch up the client to the source by speeding up the audio. So, do we want this?

@Lakshya2610 Is this happening on this new web app only or did we have this issue before also? I didn't rework the code behind webrtc and the stream so not sure if this issue has to do with anything in this PR. If not then maybe we can open a separate issue for that.

File: client-redesign/src/components/roomName.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (alir128) [reply client-redesign/src/components/roomName.vue]: I will try to use Vue plugin functionality to store webRTC calls and sockets.

**Pull request merged**